core/lib/multivm/src/tracers/prestate_tracer/vm_1_4_1/mod.rs

```diff
@@ -0,0 +1,66 @@
+use zk_evm_1_4_1::tracing::{BeforeExecutionData, VmLocalStateData};
+use zksync_state::{StoragePtr, WriteStorage};
+use zksync_types::StorageKey;
+
+use super::{
+    get_account_data, process_modified_storage_keys, process_result, PrestateTracer, State,
+    StorageAccess,
+};
+use crate::{
+    interface::dyn_tracers::vm_1_4_1::DynTracer,
+    tracers::prestate_tracer::U256,
+    vm_1_4_1::{BootloaderState, HistoryMode, SimpleMemory, VmTracer, ZkSyncVmState},
+};
+impl<S: WriteStorage, H: HistoryMode> DynTracer<S, SimpleMemory<H>> for PrestateTracer {
+    fn before_execution(
+        &mut self,
+        _state: VmLocalStateData<'_>,
+        _data: BeforeExecutionData,
+        _memory: &SimpleMemory<H>,
+        storage: StoragePtr<S>,
+    ) {
+        if self.config.diff_mode {
+            self.pre
+                .extend(process_modified_storage_keys(self.pre.clone(), &storage));
+        }
+    }
+}
+
+impl<S: WriteStorage, H: HistoryMode> VmTracer<S, H> for PrestateTracer {
+    fn after_vm_execution(
+        &mut self,
+        state: &mut ZkSyncVmState<S, H>,
+        _bootloader_state: &BootloaderState,
+        _stop_reason: crate::interface::tracer::VmExecutionStopReason,
+    ) {
+        let modified_storage_keys = state.storage.storage.inner().get_modified_storage_keys();
+        if self.config.diff_mode {
+            self.post = modified_storage_keys
+                .clone()
+                .keys()
+                .copied()
+                .collect::<Vec<_>>()
```

Comment:
You don't need to collect vector here, just iter over keys suppose to be sufficient ( and don't forget to change it for all implementations)